src/cswinrt/code_writers.h

```diff
@@ -8266,31 +8266,16 @@ return global::System.Runtime.InteropServices.CustomQueryInterfaceResult.NotHand
         auto type_name = write_type_name_temp(w, type, "%", typedef_name_type::CCW);
         auto wrapped_type_name = write_type_name_temp(w, type, "%", typedef_name_type::Projected);
         auto default_interface_name = get_default_interface_name(w, type, false, true);
-        auto base_semantics = get_type_semantics(type.Extends());
-        auto from_abi_new = !std::holds_alternative<object_type>(base_semantics) ? "new " : "";
 
+        // This type can be empty, as it is only used for metadata lookup, but not as implementation
         w.write(R"(%%[global::WinRT.ProjectedRuntimeClass(typeof(%))]
-%internal % partial class %%
-{
-public %(% comp)
+%internal % partial class %
```

Comment:
It is looking like we are going to need to a fallback where we have the interfaces for .NET Standard (given we don't generate the winrt exposed type attribute with the vtables for it today).  We happen to have someone relying on that support it seems.  On .NET Standard, we can choose to either generate this as an abstract class with the interfaces implemented with no content or we can just do if else where on .NET Standard we have the old logic and else we have your new generation code.

Reply:
Kept the original code to reduce risk and make things simpler, done! 🙂